tools/apidiff/merger.cs

```diff
@@ -76,22 +84,29 @@ public static void Process (string platform, string path, string os)
 		headers.WriteLine ($"# {title}");
 		headers.WriteLine ();
 
-		File.WriteAllText (filename, headers.ToString ());
+		var filePath = DestinationPath is not null ? Path.Combine (DestinationPath, filename) : filename;
```

Comment:
DestinationPath input has not been checked and that is going to be problematic. For example, if the uses provides a destination path is a nested directory and does not exists:
```csharp
DestinationPath = "/This/path/exists/up/to/here/the/rest/is/missing";
var filePath = Path.Combine (DestionationPath, "file.txt");
Fire.WriteAllText (filePath, "data");
```
This is going to throw an exception later of the type:
```
System.IO.DirectoryNotFoundException: Could not find a part of the path "/Users/mandel/Desktop/test/path/missing/test.txt".
  at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize, System.Boolean anonymous, System.IO.FileOptions options) [0x00164] in <4734b36ef0d2496fb9a068f0457cfdd2>:0
  at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize, System.IO.FileOptions options) [0x00000] in <4734b36ef0d2496fb9a068f0457cfdd2>:0
  at (wrapper remoting-invoke-with-check) System.IO.FileStream..ctor(string,System.IO.FileMode,System.IO.FileAccess,System.IO.FileShare,int,System.IO.FileOptions)
  at System.IO.StreamWriter..ctor (System.String path, System.Boolean append, System.Text.Encoding encoding, System.Int32 bufferSize) [0x00055] in <4734b36ef0d2496fb9a068f0457cfdd2>:0
  at System.IO.StreamWriter..ctor (System.String path) [0x00008] in <4734b36ef0d2496fb9a068f0457cfdd2>:0
  at (wrapper remoting-invoke-with-check) System.IO.StreamWriter..ctor(string)
  at System.IO.File.WriteAllText (System.String path, System.String contents) [0x00026] in <4734b36ef0d2496fb9a068f0457cfdd2>:0
  at <InteractiveExpressionClass>.Host (System.Object& $retval) [0x00000] in <df21e27c9dfc4f0f9b43bb324193a617>:0
  at Mono.CSharp.Evaluator.Evaluate (System.String input, System.Object& result, System.Boolean& result_set) [0x00038] in <af26d935fe594d42a0bac880bdce2608>:0
  at Mono.CSharpShell.Evaluate (System.String input) [0x00000] in <e263b912ccc44ce9821ebbcb30b01d36>:0
```

If the path does not exist, you can create it and  use it. Or if we prefer, raise an exception giving the path for easier debugging.

Reply:
Good call!